crates/doc/src/reduce/strategy.rs

```diff
@@ -301,6 +309,54 @@ impl Strategy {
         }
     }
 
+    fn json_schema_merge<'alloc, L: AsNode, R: AsNode>(
+        cur: Cursor<'alloc, '_, '_, '_, '_, L, R>,
+    ) -> Result<HeapNode<'alloc>> {
+        let Cursor {
+            tape,
+            loc,
+            full: _,
+            lhs,
+            rhs,
+            alloc,
+        } = cur;
+
+        let (lhs, rhs) = (lhs.into_heap_node(alloc), rhs.into_heap_node(alloc));
```

Comment:
nit: i believe you could alternately work directly with `Node` and avoid mapping into HeapNode.

This is so minor, though, to not even be worth the effort if this already works.

Reply:
I tried to do this at first and it ended up being a bigger change than I wanted so I think I'm gonna leave it be -- `lhs` and `rhs` are `LazyNode`s which don't directly implement `AsNode` and I wasn't able to easily implement it because of the generic in `LazyNode::Node`. I feel like it should be possible since `lazy_node.as_node()` would to have a clearly defined type in all scenarios: whatever `N` is when you have `LazyNode::Node`, and `HeapNode` if you have `LazyNode::Heap`.